common/pkg/libartifact/store/reference_test.go

```diff
@@ -0,0 +1,186 @@
+package store
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.podman.io/image/v5/types"
+)
+
+func TestNewArtifactReference(t *testing.T) {
```

Comment:
(Consider using a table-driven test.)

Reply:
(This is a non-blocking comment either way. Having verbose tests is better than not having tests.)

I don’t care that much about the structure… the valuable part is avoiding repetition in the common parts.
- As a minimal example, there is `image/docker/docker_transport_test.go:testParseReference`.
- It might be easier to have a separate table+loop for successes and for failures, situation-specific
- There is now `testing.T.Run`, which causes every item to show up in the test output. Situation-specific again.